sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,851 @@
+package docker
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	mathrand "math/rand"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/internalshared/reloadutil"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/testing/stepwise"
+	"golang.org/x/net/http2"
+
```

Comment:
I've been suggesting the removal of this extra newline, but are we separating named imports on purpose?

Reply:
I don't believe so. My editor will keep them separate if they start separate but otherwise add the named imports and sort in alpha order. Personally I like the separation, calling out imported packages that are given names:

```go
import (
	"encoding/base64"
	"fmt"
	"os"
	"testing"

	"github.com/hashicorp/vault/api"
	"github.com/hashicorp/vault/sdk/helper/keysutil"
	"github.com/hashicorp/vault/sdk/testing/stepwise"
	"github.com/mitchellh/mapstructure"

	dockerEnvironment "github.com/hashicorp/vault/sdk/testing/stepwise/environments/docker"
)
```

vs. 

```go
import (
	"encoding/base64"
	"fmt"
	"os"
	"testing"

	"github.com/hashicorp/vault/api"
	"github.com/hashicorp/vault/sdk/helper/keysutil"
	"github.com/hashicorp/vault/sdk/testing/stepwise"
	dockerEnvironment "github.com/hashicorp/vault/sdk/testing/stepwise/environments/docker"
	"github.com/mitchellh/mapstructure"
)
``` 

That said, I'm fine accepting the suggestions and having them in with the others, I just think the separation provides more clarity.

Reply:
Yeah, my editor does so too (it separates them). Let keep things consistent across the repo and combine them for now. We can pitch this to the team more widely later to establish convention around this.

Reply:
Rearraged in 5467f2561